test/Tests/src/Data/Text_Spec.enso

```diff
@@ -1116,90 +1124,89 @@ spec =
             accents = 'a\u{301}e\u{301}o\u{301}'
-            accents.locate accent_1 Matching_Mode.First matcher=regex . should_equal (Span.Value (1.up_to 2) accents)
+            accents.match accent_1 Matching_Mode.First matcher=regex . should_equal 'e\u{301}'
+
         Test.specify "should correctly handle regex edge cases in locate" pending="Figure out how to make Regex correctly handle empty patterns." <|
             regex = Regex_Matcher.Value
-            "".locate "foo" matcher=regex . should_equal Nothing
-            "".locate "foo" matcher=regex mode=Matching_Mode.Last . should_equal Nothing
-            "".locate_all "foo" matcher=regex . should_equal []
-            "".locate "" matcher=regex . should_equal (Span.Value (0.up_to 0) "")
-            "".locate_all "" matcher=regex . should_equal [Span.Value (0.up_to 0) ""]
-            "".locate "" matcher=regex mode=Matching_Mode.Last . should_equal (Span.Value (0.up_to 0) "")
+            "".match "foo" matcher=regex . should_equal Nothing
+            "".match "foo" matcher=regex mode=Matching_Mode.Last . should_equal Nothing
+            "".match_all "foo" matcher=regex . should_equal []
+            "".match "" matcher=regex . should_equal ""
+            "".match_all "" matcher=regex . should_equal [""]
+            "".match "" matcher=regex mode=Matching_Mode.Last . should_equal ""
             abc = 'A\u{301}ßC'
-            abc.locate "" matcher=regex . should_equal (Span.Value (0.up_to 0) abc)
-            abc.locate_all "" matcher=regex . should_equal [Span.Value (0.up_to 0) abc, Span.Value (0.up_to 0) abc, Span.Value (1.up_to 1) abc, Span.Value (2.up_to 2) abc, Span.Value (3.up_to 3) abc]
-            abc.locate "" matcher=regex mode=Matching_Mode.Last . should_equal (Span.Value (3.up_to 3) abc)
+            abc.match "" matcher=regex . should_equal abc
+            abc.match_all "" matcher=regex . should_equal ["", "", "", "", ""]
+            abc.match "" matcher=regex mode=Matching_Mode.Last . should_equal ""
 
         Test.specify "should handle overlapping matches as shown in the examples" <|
-            "aaa".locate "aa" mode=Matching_Mode.Last matcher=Text_Matcher.Case_Sensitive . should_equal (Span.Value (1.up_to 3) "aaa")
-            "aaa".locate "aa" mode=Matching_Mode.Last matcher=Regex_Matcher.Value . should_equal (Span.Value (0.up_to 2) "aaa")
+            "aaa".locate "aa" mode=Matching_Mode.Last case_sensitivity=Case_Sensitivity.Sensitive . should_equal (Span.Value (1.up_to 3) "aaa")
 
-            "aaa aaa".locate "aa" mode=Matching_Mode.Last matcher=Text_Matcher.Case_Sensitive . should_equal (Span.Value (5.up_to 7) "aaa aaa")
-            "aaa aaa".locate "aa" mode=Matching_Mode.Last matcher=Regex_Matcher.Value . should_equal (Span.Value (4.up_to 6) "aaa aaa")
+            "aaa aaa".locate "aa" mode=Matching_Mode.Last case_sensitivity=Case_Sensitivity.Sensitive . should_equal (Span.Value (5.up_to 7) "aaa aaa")
```

Comment:
Please update these as in examples - we can retain these `locate` tests but let's also add the new `match` tests based on `aAa`.

Reply:
Done